hooks/useClickOutside.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import type { RefObject, MouseEvent, TouchEvent } from 'react';
+
+// eslint-disable-next-line no-unused-vars
+type TypeHandler = (event: MouseEvent | TouchEvent) => void;
+
+export const useClickOutside = (
+  ref: RefObject<HTMLElement>,
+  handler: TypeHandler
+) => {
+  useEffect(() => {
+    const listener = (event: MouseEvent | TouchEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) {
```

Comment:
Why a type casting needed? Also let's un-negate this condition:

```tsx
if (ref.current && !ref.current.contains(event.target)) {
  handler(event);
}
```

Reply:
with your code we have:
"event.target is not attribuable to node"

Reply:
Try `event.currentTarget`

Reply:
Good i add this to my personal cheatsheet

Reply:
I search on web and other people use `as node`